src/devices/elro_db286a.c

```diff
@@ -0,0 +1,104 @@
+/* Initial doorbell implementation for Elro DB286A devices
+ * 
+ * Note that each device seems to have two id patterns, which alternate
+ * for every other button press.
+ */
+ 
+#include "rtl_433.h"
+#include "pulse_demod.h"
+#include "data.h"
+#include "util.h"
+
+static int doorbell_db286a_callback(bitbuffer_t *bitbuffer) {
+
+	//33 pulses per data pattern
+	const unsigned pulsecount = 33;
+	//One leading zero pulse + 15*33 data pattern - 1 missing trailing zero pulse 
+	const unsigned totalpulses = 33*15;
+	//Minimum data pattern repetitions (14 is maximum)
+	const unsigned minpattern = 5;
+	
+	char time_str[LOCAL_TIME_BUFLEN];
+	data_t *data;
+	bitrow_t *bb = bitbuffer->bb;
+	uint8_t *b = bb[0];
+	unsigned bits = bitbuffer->bits_per_row[0];
+
+	if (bits != totalpulses) {
+		return 0;
+	}
+	
+	//33 pulses + trailing null byte for C string
+	//Example pattern: 001101111111011000101010011011001
+	
+	char id_string[pulsecount + 1];
+	char *idsp = id_string;
+	
+	char bitrow_string[totalpulses + 1];
+	char *brp = bitrow_string;
```

Comment:
These 2 variables that uses a variable reference might be vla's. Use a define and they will for sure be constants.